src/quicktype-core/language/CPlusPlus.ts

```diff
@@ -762,7 +912,122 @@ inline ${optionalType}<T> get_optional(const json &j, const char *property) {
             });
         }
 
-        this.ensureBlankLine();
-        this.emitLine("#endif");
+        this.finishFile();
+    }
+
+    protected emitClassDefinition(c: ClassType, className: Name): void {
+        this.startFile(this.sourcelikeToString(className)+".hpp");
+        /** 
+         * Need to generate "imports", in terms 'c' has members, which
+         * are defined by others
+         */
+        let included: boolean = false;
+        this.forEachClassProperty(c, "none", (_name, _jsonName, property) => {
+
+            let p = property.type;
+
+            if (p instanceof ArrayType) {
+                p = p.items;
+            }
+
+            const propertyType = this.sourcelikeToString(this.cppType(p, { needsForwardIndirection: true, needsOptionalIndirection: true, inJsonNamespace: false }, true));
+            if (p instanceof ClassType ||
+                p instanceof EnumType ||
+                p instanceof UnionType) {
+
+                this.forEachDeclaration("none", decl => {
+                    const t = decl.type;
+                    const declType = this.sourcelikeToString(this.cppType(t, { needsForwardIndirection: true, needsOptionalIndirection: true, inJsonNamespace: false }, true));
+                    if (t instanceof ClassType ||
+                        t instanceof EnumType ||
+                        t instanceof UnionType) {
+                        if (propertyType === declType) {
+                            const include = (name: string): void => {
+                                this.emitLine(`#include ${name}`);
+                            };
+                            include("\""+declType+".hpp\"");
+                            included = true;
+                        }
+                    }
+                });
+            }
+        });
+
+        if (included) {
+            this.ensureBlankLine();
+        }
+
+        this.emitNamespaces(this._namespaceNames, () => {
+            this.ensureBlankLine();
+            this.emitDescription(this.descriptionForType(c));
+            this.ensureBlankLine();
+            this.emitLine("using nlohmann::json;");
+            this.ensureBlankLine();
```

Comment:
Six of these lines are repeated in `emitEnumDefinition` and `emitUnionDefinition`.  They should be in a method that takes a function that only does

```
            this.emitClass(c, className);
             this.ensureBlankLine();
             this.emitClassFunctions(c, className);
```